### ISSUE TITLE
Add embedded code examples dotty website

### ISSUE THREAD

**Issue by felixmulder** [lampepfl/dotty#2684]
Depends on: https://github.com/scalacenter/scastie/issues/154

We would like to take some of the code examples from the documentation pages and put them into embedded views on the landing page.

Something like:

- [ ] Enums
- [ ] Union & Intersection types
- [ ] Trait Parameters

Would be a good start to have examples of.

**Comment by Varunram** [comment]
@felixmulder May I work on it? :)

**Comment by felixmulder** [comment]
Of course you may :)

But you **can** not, currently! The feature is not implemented yet in
Scastie. See the ticket referenced above.

**Comment by Varunram** [comment]
Yeah, saw that. Once that is merged, I'll start on this one. Meanwhile, I'll look into #2689
